engine/src/main/java/org/terasology/rendering/opengl/FBOConfig.java

```diff
@@ -0,0 +1,204 @@
+     * an FBO with no attachments or one with a single color buffer attached to it.
+     * <p>
+     * To attach additional buffers, see the use*Buffer() methods.
+     * <p>
+     * Example: FBO basicFBO = new FBObuilder("basic", new Dimensions(1920, 1080), Type.DEFAULT).build();
+     *
+     * @param title      A string identifier, the title is used to later manipulate the FBO through
+     *                   methods such as LwjglRenderingProcess.getFBO(title) and LwjglRenderingProcess.bindFBO(title).
+     * @param dimensions A Dimensions object providing width and height information.
+     * @param type       Type.DEFAULT will result in a 32 bit color buffer attached to the FBO. (GL_RGBA, GL11.GL_UNSIGNED_BYTE, GL_LINEAR)
+     *                   Type.HDR will result in a 64 bit color buffer attached to the FBO. (GL_RGBA, GL_HALF_FLOAT_ARB, GL_LINEAR)
+     *                   Type.NO_COLOR will result in -no- color buffer attached to the FBO
+     *                   (WARNING: this could result in an FBO with Status.DISPOSED - see FBO.getStatus()).
+     */
+    public FBOConfig(String title, FBO.Dimensions dimensions, FBO.Type type) {
+        this.title = title;
+        this.dimensions = dimensions;
+        this.type = type;
+        this.isStatic = false;
+        isRelativeToScreenSize = false;
+    }
+
+    /**
+     * Same as the previous FBObuilder constructor, but taking in input
+     * explicit, integer width and height instead of a Dimensions object.
+     */
+    public FBOConfig(String title, int width, int height, FBO.Type type) {
+        this(title, new FBO.Dimensions(width, height), type);
+    }
+
+    public FBOConfig(String title, float scale, FBO.Type type) {
+        Preconditions.checkArgument(scale != 0, "scale can not be zero.");
+        this.title = title;
+        this.type = type;
+        this.scale = scale;
+        this.isRelativeToScreenSize = true;
+    }
+
+    public FBOConfig(String title, float scale, FBO.Type type, boolean isStatic) {
+        this(title, scale, type);
+        this.isStatic = isStatic;
+    }
+
+    public boolean isFBODynamic() {
+        return !this.isStatic;
+    }
+/*
+ *  * @param useDepthBuffer If true the FBO will have a 24 bit depth buffer attached to it. (GL_DEPTH_COMPONENT24, GL_UNSIGNED_INT, GL_NEAREST)
+    * @param useNormalBuffer If true the FBO will have a 32 bit normals buffer attached to it. (GL_RGBA, GL_UNSIGNED_BYTE, GL_LINEAR)
+    * @param useLightBuffer If true the FBO will have 32/64 bit light buffer attached to it, depending if Type is DEFAULT/HDR.
+*                       (GL_RGBA/GL_RGBA16F_ARB, GL_UNSIGNED_BYTE/GL_HALF_FLOAT_ARB, GL_LINEAR)
+    * @param useStencilBuffer If true the depth buffer will also have an 8 bit Stencil buffer associated with it.
+    *                         (GL_DEPTH24_STENCIL8_EXT, GL_UNSIGNED_INT_24_8_EXT, GL_NEAREST)
+                *                         */
```

Comment:
Further note on comments, wait until we fully discussed the architecture, but if this class survives it will need a thorough double-checking of its comments and javadoc, as they are still using terms such as FBOBuilder and create() and build().